### ISSUE TITLE
Requires interface doesn't get brought down when parent is taken down

### ISSUE THREAD

**Issue by rany2** [ifupdown-ng/ifupdown-ng#214]
Consider the following setup:

```
auto if1
iface if1
  >>some config here<<

auto if2
iface if2
  requires if1
  >> some config here <<
```

When if1 is brought down, if2 doesn't get brought down before hand breaking the network. 

Also if if1 is brought back up if2 should go up as well.

**Comment by rany2** [comment]
I thought I could work around it by just having `ifdown` & `ifup` calls in pre-down/post-up but it doesn't work. Is there some more other workaround?

**Comment by kaniini** [comment]
> When if1 is brought down, if2 doesn't get brought down before hand breaking the network.

Agreed that this is the desired behavior but there are tests for it.

> Also if if1 is brought back up if2 should go up as well.

The preference should be added as an interface-specific preference.  It does not automatically follow that I want if2 to be up if I brought up if1.  Maybe if the interface is marked as auto?
